spec/features/shared_examples/cherrypicking.rb

```diff
@@ -237,14 +237,27 @@
 
             click_link("Download #{robot.name} File")
 
+            current_destination_plate = Plate.find_by_barcode(destination_barcode)
+
             # robot file generation differs by generator
             case robot.generation_behaviour_property.value
             when 'Hamilton'
               # for Robot::Generator::Hamilton
               generated_file = DownloadHelpers.downloaded_file("#{batch_id}_batch_#{destination_barcode}_#{pick_number_index}.csv")
               generated_lines = generated_file.lines
 
-              # optionally if an expected file was supplied. compare it to the result
+              # check generated lines match expected by calculation - headers row + one row per sample/control transfer
+              expected_num_lines = current_destination_plate.wells.map(&:transfer_requests_as_target).count + 1
```

Comment:
why do you only check num_lines for the Hamilton?

Reply:
Because calculating it for the complex Tecan files is far from straight forward.
The Tecan has an additional 4 buffer lines (3 text one blank) for any destination wells that require buffer, which depends on what concentrations you've set in your test for each of the source samples.
Looking at it next but releasing what I've done so far for the UAT release.

Reply:
Hamilton is much easier as the buffer is incorporated into the same sample row.